Adding the correct working syntax for known hosts (#615)
astronomer/docs: software/deploy-git-sync.md

    - **DAGs Directory:** The directory in your Git repository that hosts your DAGs. Specify the directory's path as relative to the repository's root directory. To use your root directory as your DAGs directory, specify this value as `./`. Other changes outside the DAGs directory in your Git repository must be deployed using `astro deploy`
    - **Rev**: The commit reference of the branch that you want to sync with your Deployment
    - **Ssh Key**: The SSH private key for your Git repository

>     - **Known Hosts**: The known_hosts for your Git provider, formatted as: `github.com <github_public_key>`.In order to generate github_public_key, Kindly refer to Airflow production [guide](https://airflow.apache.org/docs/helm-chart/stable/production-guide.html#production-guide-knownhosts) for known hosts. We need to make sure to generate the ssh keys without a passphrase, use the same key type & add your corresponding SSH public key in the GitHub settings section.

Reviewer: ```suggestion
    - **Known Hosts**: The known_hosts for your Git provider, formatted as: `github.com <github_public_key>`. To generate a `github_public_key`, refer to [Apache Airflow documentation on knownHosts](https://airflow.apache.org/docs/helm-chart/stable/production-guide.html#production-guide-knownhosts). To generate the SSH key without a passphrase, use the same key type & add your corresponding SSH public key in the GitHub settings section.
```

Reviewer: @nandlalyadav57 Thank you for this! Can you clarify or expand on what you mean here:

> We need to make sure to generate the ssh keys without a passphrase, use the same key type & add your corresponding SSH public key in the GitHub settings section.

- By passphrase, do you mean password?
- What is the key type?
- Which step are you referring to in Airflow docs? 

Once we make this clear in docs, happy to approve and merge! This sounds like helpful information.

Reviewer 2: `known_hosts` file entries do not have passphrases because they are public keys. Only private keys have passphrases. public keys are by their nature public, and do not need to be protected.

I think we should avoid using `github.com`, or any specific git host, unless we are going to give a very specific example that works for that provider.

It might be worth showing them how to get this info using `ssh-keyscan`:

```
$ ssh-keyscan github.com 2>/dev/null
github.com ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBEmKSENjQEezOmxkZMy7opKgwFB9nkt5YRrYMjNuG5N87uRgg6CLrbo5wAdT/y6v0mKV0U2w0WZ2YB/++Tpockg=
github.com ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ==
github.com ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIOMqqnkVzrm0SdG6UOoqKLsabgH5C9okWi0dh2l9GKJl
```

Reviewer 3: - though known hosts don't have passphrases in the doc it speaks about generating the ssh key and that should be generated without a password 

- the type shouldnt matter and i do believe daniels approach is better (essentially showing how to get the information)

Author: **By passphrase, do you mean password?**   
The passphrase is used for encrypting the key so that it cannot be used even if someone obtains the private key file. 
We just need to press enter twice i.e.empty for no passphrase

e.g below:

`ssh-keygen -t ed25519 -C "test"`

_Generating public/private ed25519 key pair.
Enter file in which to save the key (/root/.ssh/id_ed25519):
Enter passphrase (empty for no passphrase):
Enter same passphrase again:
Your identification has been saved in /root/.ssh/id_ed25519.
Your public key has been saved in /root/.ssh/id_ed25519.pub.
The key fingerprint is:
SHA256:PhcV+BAB9NK9w2iBaEUvKbCH7NZLQvrdA3xkJKbGiB0 test
The key's randomart image is:
+--[ED25519 256]--+
|  E . oo*.o+.    |
| o = * = *o. .   |
|. o O = B =oo    |
|   = = + o =..   |
|  . + = S + +    |
|   o + * . . .   |
|    . o = .      |
|         +       |
|                 |
+----[SHA256]-----+_

Once Public and private keys are generated. Add the ''id_ed25519.pub'' public SSH key to your Github account settings and add the id_ed25519 private key on the ssh-key field on UI

**What is the key type?**      

RSA, DSA, and ECDSA

Either of the algorithms can be selected using the -t option 

ssh-rsa , ecdsa-sha2-nistp256 , ssh-ed25519

**Which step are you referring to in Airflow docs?** 

 same as in the link

generate known_hosts for github.com
`ssh-keyscan github.com > github_public_key`

get file input
`cat github_public_key `

or
above command would work too `ssh-keyscan github.com 2>/dev/null`

Reviewer 2: > A passphrase is a sequence of words or other text used to control access to a compute system, program or data. It is similar to a password in usage, but a passphrase is generally longer for added security. - <https://en.wikipedia.org/wiki/Passphrase>

In the context of ssh keys, "password" and "passphrase" are pretty interchangeable. It's the secret string you use to unlock the private key. It could be a word, it could be a phrase, but it's a string of characters that is secret.

My guess is that ssh uses "passphrase" to describe the secret used to unlock the private key in order to distinguish it from the password that is used to log into the target system.

Reviewer 4: Okay based on the discussion here I updated @paolaperaza 's original suggestion based on a few assumptions:

- The Airflow docs refer to the entire string (domain) included, as the "public key", so I removed the existing formatting and direct people to copy/paste whatever comes out when they run the listed command.
- I removed all references to passwords because this is a public key.
- Removing GitHub-only steps and instead leaving that in an example section.

Reviewer 3: its good though `ssh-keyscan -t rsa` i believe would only work if you used rsa to generate your key

Reviewer 2: No, `ssh-keyscan -t rsa` tells ssh-keyscan to only fetch the rsa public key of the server. IE: only the RSA key from the following list:

```
$ find /etc/ssh/ -iname '*.pub'
/etc/ssh/ssh_host_ecdsa_key.pub
/etc/ssh/ssh_host_rsa_key.pub
/etc/ssh/ssh_host_ed25519_key.pub
```

The user's public key doesn't come into the picture.

We keep coming back to mixing up the server's public key with the user's auth key pair. Maybe we should make the docs more explicitly point out that these are two completely different things that happen to use the same public/private key mechanism.

Reviewer 2: Here's another insight into what is going on:

```
$ sudo cat /etc/ssh/ssh_host_rsa_key.pub
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDjyCLI1H7AzOT1VpplrNeHuhguoIgJoXz2diA9LrWb1+Y4FiqeaVcx02Qy7XiRLDq9GrLYJXZTfHOTOKra7SHkg862+5cFIrXK3/ZdOR2/MEmpi/6lhNJP7VxfV9F+UecFQ8Vf7uthi/qThpQYYSG/A4B5smIyzwoXJgVI6CnlMSi3H2cmeA3UlcNmeCBd3lar8762I2g4BKyLL5BBrs4T1RSotu17ZVSO1Up7JEeyXDC87GLZQFe+Y9klmbXFFs7/ZaQEgq+etHnPQP7NcRmwgmFHw9RCtSVk/xI6JQx6CBdFXzJ+lkHte1ViVX5LqtaZEQAM8Eq+dUzSELKWJSc1 root@litten
$ ssh-keyscan -t rsa localhost 2>/dev/null
localhost ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDjyCLI1H7AzOT1VpplrNeHuhguoIgJoXz2diA9LrWb1+Y4FiqeaVcx02Qy7XiRLDq9GrLYJXZTfHOTOKra7SHkg862+5cFIrXK3/ZdOR2/MEmpi/6lhNJP7VxfV9F+UecFQ8Vf7uthi/qThpQYYSG/A4B5smIyzwoXJgVI6CnlMSi3H2cmeA3UlcNmeCBd3lar8762I2g4BKyLL5BBrs4T1RSotu17ZVSO1Up7JEeyXDC87GLZQFe+Y9klmbXFFs7/ZaQEgq+etHnPQP7NcRmwgmFHw9RCtSVk/xI6JQx6CBdFXzJ+lkHte1ViVX5LqtaZEQAM8Eq+dUzSELKWJSc1
```

Those are the same thing, except `ssh-keyscan` prepends the hostname "localhost" and leaves off the key's comment at the end "root@litten"

Reviewer 3: so then that command should work then @danielhoherd and we should be good?

>     - **Known Hosts**: The known_hosts for your Git provider, formatted as: `github.com <github_public_key>`.In order to generate github_public_key, Kindly refer to Airflow production [guide](https://airflow.apache.org/docs/helm-chart/stable/production-guide.html#production-guide-knownhosts) for known hosts. We need to make sure to generate the ssh keys without a passphrase, use the same key type & add your corresponding SSH public key in the GitHub settings section.

Reviewer 2: ```suggestion
    - **Known Hosts**: The known_hosts for your git provider, formatted as: `<hostname> <git_server_public_key>`.In order to generate git_server_public_key, refer to Airflow production [guide](https://airflow.apache.org/docs/helm-chart/stable/production-guide.html#production-guide-knownhosts) for known hosts.
```
